File: C++ WorldSimulatorGame/POproj1/Punkt.cpp
#include "Punkt.h"

Punkt::Punkt() {
	this->x = 0;
	this->y = 0;
}

Punkt::Punkt(int x, int y) {
	this->x = x;
	this->y = y;
}

int Punkt::getX() {
	return x;
}
int Punkt::getY() {
	return y;
}
void Punkt::setX(int x) {
	this->x = x;
}
void Punkt::setY(int y) {
	this->y = y;
}

bool Punkt::operator==(const Punkt& other) {
	return (x == other.x && y == other.y);
}
bool Punkt::operator!=(const Punkt& other) {
	return (!(x == other.x && y == other.y));
}
Punkt& Punkt::operator=(const Punkt& right) { // kopiujacy operator przypisania
	this->x = right.x;
	this->y = right.y;
	return *this;
}

File: Python WorldSimulatorGame/Punkt.py
import math

class Punkt:

    def __init__(self, x: int = 0, y: int = 0):
        self.x = x
        self.y = y

    def printPunkt(self):
        return str(self.x) + " " + str(self.y)

    def setPunkt(self, p):
        self.x = p.x
        self.y = p.y

    def setX(self, x):
        self.x = x

    def setY(self, y):
        self.y = y

    def getX(self):
        return self.x

    def getY(self):
        return self.y

    def sprwadzCzyTenSam(self, other):
        if other.x == self.x and other.y == self.y:
            return True
        return False

    def getDystans(self, other):
        return math.sqrt(pow(abs(self.x - other.x), 2) + pow(abs(self.y - other.y), 2))

